src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeColorization.cs

```diff
@@ -0,0 +1,199 @@
+
+    #endregion
+
+    #region Constructor
+
+    static ThemeColorization()
+    {
+        foreach (ResourceDictionary mergedDictionary in Application.Current.Resources.MergedDictionaries)
+        {
+            if(mergedDictionary.Source != null && mergedDictionary.Source.ToString().EndsWith(_themeDictionaryName))
+            {
+                _isFluentWindowsThemeEnabled = true;
+                break;
+            }
+        }
+
+        _currentApplicationTheme = GetSystemTheme();
+    }
+
+    #endregion
+
+    // ------------------------------------------------
+    //
+    // Methods
+    //
+    // ------------------------------------------------
+    #region internal Methods
+
+    internal static bool IsFluentWindowsThemeEnabled
+    {
+        get { return _isFluentWindowsThemeEnabled; }
+    }
+    
+    internal static void UpdateApplicationTheme()
+    {
+        string themeToApply = GetSystemTheme();
+        string appsThemeToApply = GetAppsTheme();
+
+        Color currentApplicationAccentColor = DWMColorization.CurrentApplicationAccentColor;
+        Color accentColorToApply = DWMColorization.GetSystemAccentColor();
+
+        if (themeToApply != _currentApplicationTheme || accentColorToApply != currentApplicationAccentColor || appsThemeToApply != _currentAppsTheme)
+        {
+            DWMColorization.UpdateAccentColors();
+            ApplyTheme();
+        }
+    }
+
+    /// <summary>
+    /// Updates the application resources with the specified <see cref="ResourceDictionary"/>.
+    /// </summary>
+    /// <param name="newDictionary">The new <see cref="ResourceDictionary"/> to update the application resources with.</param>
+    /// <exception cref="ArgumentNullException">Thrown when <paramref name="newDictionary"/> is null.</exception>
+    internal static void UpdateApplicationResources(Uri dictionaryUri)
+    {
+        if (dictionaryUri == null)
+        throw new ArgumentNullException(nameof(dictionaryUri));
+
+        ResourceDictionary newDictionary = new ResourceDictionary();
+        newDictionary.Source = dictionaryUri;
+
+        foreach (var key in newDictionary.Keys)
+        {
+            if (Application.Current.Resources.Contains(key))
+            {
+                if (!object.Equals(Application.Current.Resources[key], newDictionary[key]))
+                {
+                    Application.Current.Resources[key] = newDictionary[key];
+                }
+            }
+            else
+            {
+                Application.Current.Resources.Add(key, newDictionary[key]);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Updates the value of resources based on Type of theme applied in resource dictionary
+    /// </summary>
+    internal static void ApplyTheme()
+    {
+        string themeToApply = GetSystemTheme();
+
+        Window currentWindow = Application.Current.MainWindow;
+
+        if (IsThemeDark() && Utility.IsOSWindows11OrNewer)
```

Comment:
This switch (and the whole need to know theme name, which can also be retrieved using GetCurrentThemeName API) is questionable. Why do we need separate themes for the various high contrast themes? Can it not be just one using the system colors as per the [guidance](https://learn.microsoft.com/en-us/windows/apps/design/accessibility/high-contrast-themes#best-practices)?

Reply:
I haven't fully groked the code here, but just want to throw out there for everyone that High Contrast mode != High Contrast themes, introduced in Windows 11.

Reply:
@riverar can you point to some source for this ? Or can you explain a bit more about high contrast mode vs theme.

Reply:
Hi @dipeshmsft, there's some hint of this at https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-systemparametersinfoa#remarks.

But in my quick testing, it appears `SystemParametersInfo(SPI_GETHIGHCONTRAST)` is now returning a flag with `HCF_HIGHCONTRASTON` bit set, when a Contrast Theme (Windows 11) is active. (Tested on 26080.) So `SystemParameters.HighContrast` might be good enough for detecting all high contrast states, which is great. (At runtime, you can look for `WM_SETTINGCHANGE` (WindowsThemeElement) as we do in our app.)

Not saying any action is needed here, I haven't had time to review the code. But just sharing my notes for future review/work.